src/mpi_native/mpi_native.cpp

```diff
@@ -9,6 +9,7 @@
 using namespace faabric::executor;
 
 static faabric::scheduler::MpiContext executingContext;
+static std::string logName;
```

Comment:
Have you actually tested this? Each MPI rank is a thread, but this is a single global variable, won't it keep getting overwritten when each rank sets it? It should be `thread_local`-ised.

Having the logger name as the global variable seems odd, why not just a shared pointer to the logger? I would do something like:

```
static thread_local std::shared_ptr<spdlog::logger> mpiLogger = nullptr;

std::shared_ptr<spdlog::logger> getMpiLogger(int mpiRank) {
    if(mpiLogger == nullptr) {
        mpiLogger = faabric::util::getLogger(fmt::format("MPI-{}", mpiRank));        
    }
    return mpiLogger;
}
```

Then just call `getMpiLogger(rank)` inline.